contracts/FIFOMarket.sol

```diff
@@ -65,20 +77,55 @@ contract FIFOMarket is
     );
     _queueHeadIndex = 0;
     _queueNextInsertIndex = 0;
+    priorityRestrictedThreshold = 0;
+    totalSupply = 0;
+    _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+    _grantRole(ALLOWLIST_ROLE, _msgSender()); // todo, do this here or have to do it explicitly after intialization?
```

Comment:
AFAIK these all have to happen elsewhere unless we redeploy proxy (not opposed to re-deploying this contract)

Reply:
This would likely change with the new [reinitializer](https://forum.openzeppelin.com/t/release-candidate-for-contracts-4-6-open-review-period/27075) feature OZ contracts 4.6 adds

Reply:
Gonna summarize what I understand here from our convos and my research:
- When working with proxies, initializers only get called ONCE ever, unless you completely redeploy the proxy.  For this reason, making a change like this to the contract forces us to re-deploy the proxy in order for these roles to actually be granted - it's not enough to upgrade the implementation. 
- In this case, we're going to have to redeploy the proxy anyway because we're going to make breaking changes to the contract's storage anyway and everything is under development anyway, so we will just do this and it's fine.
- In fact, there's no other (secure) way to allow granting of these roles if we don't at LEAST establish a `DEFAULT_ADMIN_ROLE` during initialization. The incumbent initializer never granted that role to any address, and this contract also doesn't have a notion of "owner" or other privileged status that we could use to secure something like an external role-granting function.  So we HAVE to redeploy the proxy and grant at least that `DEFAULT_ADMIN_ROLE` role to start taking advantage of the access control features inherited by this contract.
- no other changes are required because we can use `grantRole`, `revokeRole` etc. which are already inherited by this contract to manage role access once the deployer at least has the `DEFAULT_ADMIN_ROLE`.  We might as well allowlist the deployer at the same time, but we can allowlist other addresses using `grantRole` as needed.

Reply:
addendum to the first bullet: OZ contracts 4.6 introduces a reinitialize function that allows for re-initializing contracts. Until then, all the above holds true :100: